### PR TITLE
PostgreSQL Schema

### PR DESCRIPTION
**Pull request by JasonWarrenUK**
- `user`
   - added automatic `created_at` timestamp
- `colour`
   - `hex` takes a hexadecimal value, so don't put the "#" in the DB
- `collection`
   - `type` column, which will allow us to:
      - track "favourite" vs "collection"
      - add extra types in the future (hence why it isn't a boolean)
   - added `is_public`
   - added `description`
   - added `created_at`
- `comment`
   - added `content`
   - added `created_at`
- `colour_collection`
   - added `order`, an integer to track what order the colours in a palette appear in

### CODE REVIEW

File: DB/schema.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  id SERIAL PRIMARY KEY,
+  hex CHAR(6) NOT NULL -- hex colour code, without the #
+);
+
+CREATE TABLE collection (
+  id SERIAL PRIMARY KEY,
+  type VARCHAR(255) NOT NULL, -- "favourite" or "collection", but I've made this a string rather than a bool so we can add more types in the future
+  user_id INTEGER REFERENCES user(id),
+  name VARCHAR(255) NOT NULL,
+  description TEXT -- user-created blurb
```
Comment (jackcasstlesjones) [inline DB/schema.sql]: missing comma

File: DB/schema.sql [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+);
+
+CREATE TABLE comment (
+  id SERIAL PRIMARY KEY,
+  user_id INTEGER REFERENCES user(id),
+  collection_id INTEGER REFERENCES collection(id),
+  content TEXT NOT NULL -- the actual comment text
```
Comment (jackcasstlesjones) [inline DB/schema.sql]: and here

File: DB/schema.sql [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(255) NOT NULL,
+  email VARCHAR(255) UNIQUE NOT NULL,
+  hash VARCHAR(255) NOT NULL,
+  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP -- when the user was created
+);
+
+CREATE TABLE colour (
+  id SERIAL PRIMARY KEY,
+  hex CHAR(6) NOT NULL -- hex colour code, without the #
```
Comment (jackcasstlesjones) [inline DB/schema.sql]: here

**Review by jackcasstlesjones — CHANGES_REQUESTED** [review]
missing some commas

File: DB/schema.sql (hunk 1 above)
Comment (JasonWarrenUK) [reply DB/schema.sql]: thx

File: DB/schema.sql (hunk 2 above)
Comment (JasonWarrenUK) [reply DB/schema.sql]: thx

File: DB/schema.sql (hunk 3 above)
Comment (JasonWarrenUK) [reply DB/schema.sql]: thx

**Review by jackcasstlesjones — APPROVED** [review]
LGTM

**Pull request merged**